src/net.c

```diff
@@ -753,6 +759,195 @@ static int hyper_set_interface_mtu(struct rtnl_handle *rth,
+		goto err;
+	}
+
+	if (!mac_addr && !strcmp(mac_addr, "")) {
+		fprintf(stderr, "invalid mac_addr\n");
+		goto err;
+	}
+
+	sock = socket(PF_INET, SOCK_DGRAM, 0);
+	if (sock == -1) {
+		fprintf(stderr, "failed to get socket handle: %s\n",
+			strerror(errno));
+		goto err;
+	}
+
+	if (!strncpy(ifr.ifr_name, device, strlen(device) + 1)) {
+		fprintf(stderr, "strncpy failed to copy interface"
+			" name: %s\n", strerror(errno));
+		goto err;
+	}
+
+	if (ioctl(sock, SIOCGIFHWADDR, &ifr) == -1) {
+		fprintf(stderr, "ioctl SIOCGIFHWADDR failed: %s\n",
+			strerror(errno));
+		goto err;
+	}
+
+	tmp_mac_addr = hyper_ifreq_to_mac_addr(ifr);
+	if (!tmp_mac_addr) {
+		fprintf(stderr, "tmp_mac_addr is NULL\n");
+		goto err;
+	}
+
+	if (!strcasecmp(mac_addr, tmp_mac_addr)) {
+		fprintf(stderr, "device mac address found %s does not match"
+			" with the expected mac address %s\n",
+			tmp_mac_addr, mac_addr);
+		goto err;
+	}
+
+	ret = 0;
+
+err:
+	free(tmp_mac_addr);
+	close(sock);
+	return ret;
+}
+
+/*!
+ * Find network interface list, and retrieve the name of the one
+ * matching the EUI-48 hardware address provided.
+ *
+ * \param mac_addr hardware address (EUI-48) to match.
+ * \param[out] device name of the network interface matching \p mac_addr.
+ *
+ * \note In case the function succeeds, it returns 0 and \p device is
+ * filled with the right network interface name. In case the function
+ * fails, it returns -1 and \p device will point to \c NULL.
+ */
+static int hyper_get_iface_name_from_mac_addr(const char *mac_addr,
+					      char **device)
+{
+	struct ifaddrs *ifaddr, *ifa;
+	struct ifreq ifr;
+	int sock = -1;
+	char *tmp_mac_addr;
+	int ret = -1;
+
+	if (!mac_addr) {
+		fprintf(stderr, "mac address pointer is NULL\n");
+		goto err;
+	}
+
+	if (!strcmp(mac_addr, "")) {
+		fprintf(stderr, "mac address is empty\n");
+		goto err;
+	}
+
+	if (!device) {
+		fprintf(stderr, "device pointer is NULL\n");
+		goto err;
+	}
+
+	if (getifaddrs(&ifaddr) == -1) {
+		fprintf(stderr, "failed to get interface list: %s\n",
+			strerror(errno));
+		goto err;
+	}
+
+	sock = socket(PF_INET, SOCK_DGRAM, 0);
+	if (sock == -1) {
+		fprintf(stderr, "failed to get socket handle: %s\n",
+			strerror(errno));
+		goto err1;
+	}
+
+	*device = NULL;
+
+	for (ifa = ifaddr; ifa != NULL; ifa = ifa->ifa_next) {
+		if (ifa->ifa_addr == NULL) {
+			continue;
+		}
+
+		if (!strncpy(ifr.ifr_name, ifa->ifa_name,
+			    strlen(ifa->ifa_name) + 1)) {
+			fprintf(stderr, "strncpy failed to copy interface"
+				" name: %s\n", strerror(errno));
+			goto err1;
+		}
+
+		if (ioctl(sock, SIOCGIFHWADDR, &ifr) == -1) {
+			fprintf(stderr, "ioctl SIOCGIFHWADDR failed: %s\n",
+				strerror(errno));
+			goto err1;
+		}
+
+		tmp_mac_addr = hyper_ifreq_to_mac_addr(ifr);
+		if (!tmp_mac_addr) {
+			fprintf(stderr, "tmp_mac_addr is NULL\n");
+			goto err1;
+		}
+
+		if (!strcasecmp(mac_addr, tmp_mac_addr)) {
+			free(tmp_mac_addr);
+			*device = strdup((const char*) ifr.ifr_name);
```

Comment:
Hi @sboeuf 
I think before assign a new value to ```device``` you should validate that it is null, if it is then free previous value in order to avoid memory leaks